### PR TITLE
Fix for issue https://github.com/denisenkom/pytds/issues/38

### PR DESCRIPTION
**Pull request by VFLashM**
See https://github.com/denisenkom/pytds/issues/38 for details.

When return_value is accessed, tds discards all remaining rows.
It can lead to perception that procedure returned no rows.
    
This change ensures exception is raised if fetch is performed after return_value access.

### CODE REVIEW

File: tests/tests.py [hunk 1]
```diff
@@ -1450,6 +1450,8 @@ def _retval_setup(self, cur):
 CREATE PROCEDURE add_one (@input int)
 AS
 BEGIN
+    select 'a' as a
+    select 'b' as b
```
Comment (denisenkom) [inline tests/tests.py]: Why did you change this test case?  I think it was useful, it tested stored procedures without any result-sets but with return values.

Comment (VFLashM) [inline tests/tests.py]: Yes, I've changed this test because I needed slightly different one:

1. user successfully fetches first result set
2. user reads return_code (and this action erases all remaining row data)
3. user fails to read second result set with assertion saying that reading
return_code is bad

I can recreate old one as well if you prefer.

   1.


On Mon, Jan 5, 2015 at 11:21 AM, denisenkom <notifications@github.com>
wrote:

> In tests/tests.py
> <https://github.com/denisenkom/pytds/pull/43#discussion-diff-22470022>:
>
> > @@ -1450,6 +1450,8 @@ def _retval_setup(self, cur):
> >  CREATE PROCEDURE add_one (@input int)
> >  AS
> >  BEGIN
> > +    select 'a' as a
> > +    select 'b' as b
>
> Why did you change this test case? I think it was useful, it tested stored
> procedures without any result-sets but with return values.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/denisenkom/pytds/pull/43/files#r22470022>.
>



-- 
Regards,
  Valerii Lashmanov

Comment (denisenkom) [inline tests/tests.py]: Yes,  I would like to have both tests.  Please add both.

**Comment by VFLashM** [comment]
Old empty resultset procedure test restored
